src/routes/transactions/tests/preview.rs

```diff
@@ -0,0 +1,141 @@
+use std::time::Duration;
+
+use mockall::predicate::eq;
+use rocket::http::{ContentType, Header, Status};
+use rocket::local::asynchronous::Client;
+use serde_json::json;
+
+use crate::common::models::backend::chains::{
+    BlockExplorerUriTemplate, ChainInfo, GasPrice, NativeCurrency, RpcAuthentication, RpcUri, Theme,
+};
+use crate::config::contract_info_request_timeout;
+use crate::providers::info::*;
+use crate::routes::contracts::models::DataDecoderRequest;
+use crate::routes::transactions::handlers::preview::TransactionPreview;
+use crate::routes::transactions::tests::PREVIEW_RESPONSE;
+use crate::tests::main::setup_rocket;
+use crate::utils::errors::ApiResult;
+use crate::utils::http_client::{MockHttpClient, Request, Response};
+
+fn get_chain_info() -> ChainInfo {
+    ChainInfo {
+        recommended_master_copy_version: "1.1.1".to_string(),
+        transaction_service: "https://safe-transaction.mainnet.gnosis.io".to_string(),
+        vpc_transaction_service: "http://mainnet-safe-transaction-web.safe.svc.cluster.local"
+            .to_string(),
+        chain_id: "1".to_string(),
+        chain_name: "".to_string(),
+        short_name: "eth".to_string(),
+        l2: false,
+        description: "Random description".to_string(),
+        rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        safe_apps_rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        public_rpc_uri: RpcUri {
+            authentication: RpcAuthentication::ApiKeyPath,
+            value: "".to_string(),
+        },
+        block_explorer_uri_template: BlockExplorerUriTemplate {
+            address: "".to_string(),
+            tx_hash: "".to_string(),
+            api: "".to_string(),
+        },
+        native_currency: NativeCurrency {
+            name: "".to_string(),
+            symbol: "".to_string(),
+            decimals: 0,
+            logo_uri: "https://test.token.image.url".to_string(),
+        },
+        theme: Theme {
+            text_color: "#fff".to_string(),
+            background_color: "#000".to_string(),
+        },
+        ens_registry_address: None,
+        gas_price: vec![GasPrice::Fixed {
+            wei_value: "1000000".to_string(),
+        }],
+        disabled_wallets: vec![],
+        features: vec![],
+    }
+}
+
+#[rocket::async_test]
+async fn post_preview_success() {
+    let mut mock_info_provider = MockInfoProvider::new();
+    let data = "0x095ea7b3000000000000000000000000ae9844f89d98c150f5e61bfc676d68b4921559900000000000000000000000000000000000000000000000000001c6bf52634000";
+    mock_info_provider
+        .expect_chain_info()
+        .returning_st(move || Ok(get_chain_info()));
+
+    let mock_http_client = {
+        let mut mock_http_client = MockHttpClient::new();
```

Comment:
I assume the advantage of assigning the return value of this block is that we can limit the `mut`?

Reply:
Yes. The setup is done inside the block (mimicking a builder pattern) and then we expose it as non-mutable reference.

For the test case it might not be required that we are this strict since the scope of the mutable value is just the test function itself.

I'd push it more for readability in this case (ie.: if it makes the test easier to read or not)